oxide/lib.go

```diff
@@ -68,8 +98,20 @@ func NewClient(cfg *Config) (*Client, error) {
 		Timeout: 600 * time.Second,
 	}
 
+	errs := make([]error, 0)
+
 	// Layer in the user-provided configuration if provided.
 	if cfg != nil {
+		if cfg.Profile != "" || cfg.UseDefaultProfile {
+			fileCreds, err := getProfile(*cfg)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("unable to retrieve profile: %w", err))
+			} else {
+				token = fileCreds.token
+				host = fileCreds.host
+			}
```

Comment:
Let's return the error here directly, if we can't retrieve the profile then we won't be able to do anything further. The host and token errors will show up, but they will be a direct result of the profile not being retrieved rather than two separate errors that we want to show in conjunction.

```suggestion
			if err != nil {
				return nil, fmt.Errorf("unable to retrieve profile: %w", err)
			}

			token = fileCreds.token
			host = fileCreds.host
```

Reply:
👍 done in 0fca9c247f7688d4c2879fa986ad00a0bc8f8a69